feature/gribi/encap_frr/README.md

```diff
@@ -0,0 +1,492 @@
+                                network-instance: "ENCAP_TE_VRF_A"
+                            }
+                        }
+                        rule {
+                            sequence-id: 14
+                            ipv6 {
+                                dscp-set: [dscp_encap_a_1, dscp_encap_a_2]
+                            }
+                            action {
+                                network-instance: "ENCAP_TE_VRF_A"
+                            }
+                        }
+                        rule {
+                            sequence-id: 15
+                            ipv4 {
+                                dscp-set: [dscp_encap_b_1, dscp_encap_b_2]
+                            }
+                            action {
+                                network-instance: "ENCAP_TE_VRF_B"
+                            }
+                        }
+                        rule {
+                            sequence-id: 16
+                            ipv6 {
+                                dscp-set: [dscp_encap_b_1, dscp_encap_b_2]
+                            }
+                            action {
+                                network-instance: "ENCAP_TE_VRF_B"
+                            }
+                        }
+                        rule {
+                            sequence-id: 17
+                            action {
+                                network-instance: "DEFAULT"
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+*   Using gRIBI, install the following gRIBI AFTs, and validate the specified
+    behavior.
+
+```
+IPv4Entry {138.0.11.0/24 (ENCAP_TE_VRF_A)} -> NHG#10 (DEFAULT VRF) -> {
+  {NH#201, DEFAULT VRF, weight:1},
+  {NH#202, DEFAULT VRF, weight:3},
+}
+NH#201 -> {
+  encapsulate_header: OPENCONFIGAFTTYPESENCAPSULATIONHEADERTYPE_IPV4
+  ip_in_ip {
+    dst_ip: "203.0.113.1"
+    src_ip: "ipv4_outer_src_111"
+  }
+  network_instance: "TE_VRF_111"
+}
+NH#202 -> {
+  encapsulate_header: OPENCONFIGAFTTYPESENCAPSULATIONHEADERTYPE_IPV4
+  ip_in_ip {
+    dst_ip: "203.10.113.2"
+    src_ip: "ipv4_outer_src_111"
+  }
+  network_instance: "TE_VRF_111"
+}
+
+
+IPv4Entry {203.0.113.1/32 (TE_VRF_111)} -> NHG#1 (DEFAULT VRF) -> {
+  {NH#1, DEFAULT VRF, weight:1,ip_address=192.0.2.101},
+  {NH#2, DEFAULT VRF, weight:3,ip_address=192.0.2.102},
+  backup_next_hop_group: 8 // re-encap to 203.0.113.100
+}
+IPv4Entry {192.0.2.101/32 (DEFAULT VRF)} -> NHG#2 (DEFAULT VRF) -> {
+  {NH#10, DEFAULT VRF, weight:1,mac_address:magic_mac, interface-ref:dut-port-2-interface},
+  {NH#11, DEFAULT VRF, weight:3,mac_address:magic_mac, interface-ref:dut-port-3-interface},
+}
+IPv4Entry {192.0.2.102/32 (DEFAUlT VRF)} -> NHG#3 (DEFAULT VRF) -> {
+  {NH#100, DEFAULT VRF, weight:2,mac_address:magic_mac, interface-ref:dut-port-4-interface},
+}
+
+NHG#8 (Default VRF) {
+  {NH#1000, DEFAULT VRF}
+}
+NH#1000 -> {
```

Comment:
Should this be a DecapEncap action for this NH ?
If NH#1 & NH#2 are down, then Encap traffic will be adding another Encap on existing encap header ?

Reply:
https://github.com/openconfig/featureprofiles/pull/2514

Reply:
Thanks